proxy/http/remap/NextHopHealthStatus.cc

```diff
@@ -123,7 +124,12 @@ NextHopHealthStatus::markNextHop(TSHttpTxn txn, const char *hostname, const int
           new_fail_count = h->failCount += 1;
         }
       } // end lock guard
-      NH_Note("[%" PRId64 "] NextHop %s marked as down %s", sm_id, (result.retry) ? "retry" : "initially", h->hostname.c_str());
+      if (host == nullptr) {
+        NH_Note("[%" PRId64 "] NextHop %s marked as down %s", sm_id, (result.retry) ? "retry" : "initially", h->hostname.c_str());
+      } else {
+        NH_Note("[%" PRId64 "] NextHop %s marked as down %s for request %s", sm_id, (result.retry) ? "retry" : "initially",
+                h->hostname.c_str(), host);
```

Comment:
Why not
```
        NH_Note("[%" PRId64 "] NextHop %s marked as down %s for request %s", sm_id, (result.retry) ? "retry" : "initially",
                h->hostname.c_str(), host ? host : "(null)");
```

Reply:
Because I am fairly new to C++. So have a lot to still learn. Unsure how the `?` and `:` work in that statement.

Reply:
It's a less verbose equivalent to using a lambda expression with an if statement:

`([](char const *h) -> char const * { if (h) return h; else return "(null)"; })(host)`

Reply:
Thank you for explaining, sounds like it is a `??` from Swift, where an optional can have a "default" value set. I made your suggested changes, should be updated now.